src/core/ext/client_channel/client_channel_factory.c

```diff
@@ -85,3 +90,120 @@ grpc_arg grpc_client_channel_factory_create_channel_arg(
   arg.value.pointer.vtable = &factory_arg_vtable;
   return arg;
 }
+
+struct grpc_channel_targets_info {
```

Comment:
I've used mdstr_hash_table, but behind a couple of typedefs and some wrapping functions that encapsulate the actual data structure used.